ros2action/ros2action/verb/echo.py

```diff
@@ -0,0 +1,292 @@
+            output_interface_list.update([interface.value for interface in ActionInterfaces])
+
+        if args.action_type is None:
+            with NodeStrategy(args) as node:
+                try:
+                    action_module = get_action_class(
+                        node, args.action_name)
+                except (AttributeError, ModuleNotFoundError, ValueError):
+                    raise RuntimeError(f"The action name '{args.action_name}' is invalid")
+        else:
+            try:
+                action_module = get_action(args.action_type)
+            except (AttributeError, ModuleNotFoundError, ValueError):
+                raise RuntimeError(f"The service type '{args.action_type}' is invalid")
+
+        if action_module is None:
+            raise RuntimeError('Could not load the type for the passed action')
+
+        self.csv = args.csv
+        self.truncate_length = args.truncate_length if not args.full_length else None
+        self.flow_style = args.flow_style
+        self.no_arr = args.no_arr
+        self.no_str = args.no_str
+
+        send_goal_event_topic = args.action_name + '/_action/send_goal/_service_event'
+        send_goal_event_msg_type = action_module.Impl.SendGoalService.Event
+
+        cancel_goal_event_topic = args.action_name + '/_action/cancel_goal/_service_event'
+        cancel_goal_event_msg_type = action_module.Impl.CancelGoalService.Event
+
+        get_result_event_topic = args.action_name + '/_action/get_result/_service_event'
+        get_result_event_msg_type = action_module.Impl.GetResultService.Event
+
+        feedback_topic = args.action_name + '/_action/feedback'
+        feedback_topic_type = action_module.Impl.FeedbackMessage
+
+        status_topic = args.action_name + '/_action/status'
+        status_topic_type = action_module.Impl.GoalStatusMessage
+
+        # Queue for messages from above topic
+        self.output_msg_queue = queue.Queue(args.queue_size)
+
+        run_thread = True
+        # Create a thread to output message from output_queue
+
+        def message_handler():
+            while run_thread:
+                try:
+                    message = self.output_msg_queue.get(block=True, timeout=0.5)
+                    self.output_msg_queue.task_done()
+                except queue.Empty:
+                    continue
+                print(message)
+        output_msg_thread = threading.Thread(target=message_handler)
+        output_msg_thread.start()
+
+        with NodeStrategy(args) as node:
+            send_goal_event_sub = None
+            if ActionInterfaces.GOAL_SERVICE.value in output_interface_list:
+                send_goal_event_sub: Subscription[EventMessage] = node.create_subscription(
+                        send_goal_event_msg_type,
+                        send_goal_event_topic,
+                        self._send_goal_subscriber_callback,
+                        qos_profile_services_default)
```

Comment:
this is the same QoS with `ros2 service echo`.

but actually we need to add optional argument to specify the QoS for this? because QoS can be specified by user application via `configure_introspection`. it the QoS for `configure_introspection` is strict, there is a possibility `ros2 service echo` or `ros2 action echo` cannot match with it?

i think `ros2 service echo` has the same problem, probably we can follow this up later?

Reply:
Yes, we need to provide parameters that allow users to set QoS.
I will add a TODO to indicate that the next step is to address this issue.

Reply:
sounds good to me. for the service introspection part, i think we need to backport this option.